tests/v3_api/test_project_quota.py

```diff
@@ -0,0 +1,162 @@
+    assert proj.resourceQuota.limit.pods == '20'
+    assert proj.resourceQuota.limit.requestsCpu == '2000m'
+    assert proj.namespaceDefaultResourceQuota is not None
+    assert proj.namespaceDefaultResourceQuota.limit.pods == '5'
+
+    ns = create_ns(c_client, cluster, proj)
+    print(ns)
+
+    assert ns is not None
+    assert ns.resourceQuota is not None
+    assert ns.resourceQuota.limit.pods == "5"
+    assert ns.resourceQuota.limit.requestsCpu == '500m'
+
+    validate_resoucequota_thru_kubectl(ns)
+
+
+def test_resource_quota_create_namespace_with_ns_quota():
+
+    # Create project quota and namspaces and verify
+    # namespace creation is allowed within the quota
+
+    cluster = namespace["cluster"]
+    client = get_admin_client()
+    c_client = namespace["c_client"]
+
+    quota = default_project_quota()
+    proj = client.create_project(name='test-' + random_str(),
+                                 clusterId=cluster.id,
+                                 resourceQuota=quota,
+                                 namespaceDefaultResourceQuota=quota)
+    proj = client.wait_success(proj)
+
+    assert proj.resourceQuota is not None
+
+    # Create a namespace with namespace resource quota
+    ns_name = random_str()
+    ns = c_client.create_namespace(name=ns_name,
+                                   projectId=proj.id,
+                                   resourceQuota=ns_quota())
+    ns = c_client.wait_success(ns)
+
+    assert ns is not None
+    assert ns.resourceQuota is not None
+    assert ns.resourceQuota.limit.pods == "10"
+    assert ns.resourceQuota.limit.requestsCpu == '500m'
+
+    validate_resoucequota_thru_kubectl(ns)
+
+    # Create another namespace with quota and it should succeed
+
+    ns1 = c_client.create_namespace(name=random_str(),
+                                    projectId=proj.id,
+                                    resourceQuota=ns_quota())
+    ns1 = c_client.wait_success(ns1)
+    print(ns1)
+
+    assert ns1 is not None
+    assert ns1.resourceQuota is not None
+    assert ns1.resourceQuota.limit.pods == "10"
+    assert ns1.resourceQuota.limit.requestsCpu == '500m'
+
+    validate_resoucequota_thru_kubectl(ns1)
+
+    # Creating another namespace should fail as it exceeds the allotted limit
+    try:
+        c_client.create_namespace(name=random_str(),
+                                  projectId=proj.id,
+                                  resourceQuota=ns_quota())
+    except Exception as e:
+        errorstring = str(e)
+        print(str(e))
+    assert "MaxLimitExceeded" in errorstring
+
+
+def validate_resoucequota_thru_kubectl(namespace):
+
+    # This method executes `kubectl describe quota command` and verifies if the
+    # resource quota from kubectl and the quota assigned for the namespace
+    # through API are the same
+
+    command = "describe quota --namespace " + namespace['id']
+    print(command)
+
+    result = execute_kubectl_cmd(command, json_out=False)
```

Comment:
@soumyalj Would it be more easier to get the resource quota objects when you use json_out=True for execute_kubectl_cmd ?

Reply:
Changed the code to use json=True